### ISSUE TITLE
Improve performance of ipopt-watertap for linear programs

### ISSUE THREAD

**Issue by bknueven** [watertap-org/watertap#787]
Ipopt has a few options which are best for linear programming problems; see: https://coin-or.github.io/Ipopt/OPTIONS.html and search for "LP".

Because we already do a huge amount of preprocessing in `ipopt-watertap`, it's trivial to apply these settings automatically if we're solving a LP.

**Comment by adam-a-a** [comment]
Thanks, @bknueven. To augment this effort, I think it would be useful to incorporate and document some of these options in terms of where they could aid with better initialization routines, for example.

I could see something like `least_square_init_primal` or `least_square_init_dual` helping with initialization routines where it may be difficult to arrive at a reasonable initial point:

![image](https://user-images.githubusercontent.com/22691844/193711648-4d5c788b-7e41-4677-918b-50322b418938.png)

**Comment by adam-a-a** [comment]
FYI, when I say "document" these options, I am really suggesting to document in layman's terms for WaterTAP users, providing some context on when/where certain option settings could be useful.

**Comment by bknueven** [comment]
> Thanks, @bknueven. To augment this effort, I think it would be useful to incorporate and document some of these options in terms of where they could aid with better initialization routines, for example.
> 
> I could see something like `least_square_init_primal` or `least_square_init_dual` helping with initialization routines where it may be difficult to arrive at a reasonable initial point:
> 
> ![image](https://user-images.githubusercontent.com/22691844/193711648-4d5c788b-7e41-4677-918b-50322b418938.png)

Unfortunately, most of these options are not available through the AMPL interface (which is what we use) in Ipopt 3.13 (which is what is shipped with IDAES). This was fixed in Ipopt 3.14, but in order to easily use them we'd need IDAES to update to Ipopt 3.14. This should maybe be done for other reasons, though not necessarily this one.

I think the only option relevant to LPs available through the AMPL interface in Ipopt 3.13 is the one I implement in #788, `alpha_for_y`. Because this less controversial than, say, ignoring the initialization (like `least_square_init_primal` does), I think setting that as default can't do much harm.
